### PR TITLE
Fixes #30389 - Optimze false for repo syncs

### PR DESCRIPTION
**Pull request by sjha4**
1. Create and Sync a repo with mirror set to true.
2. Re-sync the repo.
3. Contents should not get unassociated.

### CODE REVIEW

File: app/services/katello/pulp3/repository.rb [hunk 1]
```diff
@@ -194,7 +194,7 @@ def delete(href = repository_reference.try(:repository_href))
       end
 
       def sync
-        sync_url_params = {remote: repo.remote_href, mirror: repo.root.mirror_on_sync}
+        sync_url_params = {remote: repo.remote_href, mirror: repo.root.mirror_on_sync, optimize: false}
```
Comment (jlsherrill) [inline app/services/katello/pulp3/repository.rb:197]: actually, will this work for non-yum repos?  (will it be ignored)?

Comment (jlsherrill) [reply app/services/katello/pulp3/repository.rb:197]: Doesn't look like:  `optimize` is not a valid attribute in `PulpFileClient::RepositorySyncURL`. Please check the name to make sure it's valid. List of attributes: [:remote, :mirror]


I think we need to only add it if its a yum type

Comment (sjha4) [reply app/services/katello/pulp3/repository.rb:197]: Waiting for tests to fail to find out and re-record.. :)

Comment (sjha4) [reply app/services/katello/pulp3/repository.rb:197]: Yes..needs to be for only rpm..Updating..

**Comment by sjha4** [comment]
Moved this to https://github.com/Katello/katello/pull/8869 with gem update and re-recorded VCRs. Closing this.

**Pull request closed without merging**